MoviesHub/MoviesHub/Module/Home/Controller/HomeVC.swift

```diff
@@ -0,0 +1,174 @@
+        static let numberOfRowsInSection = 1
+        static let heightForHeaderInSection = 48.0
+    }
+    
+    // MARK: - Outlets
+    @IBOutlet weak private var tblHome: UITableView!
+    private let viewModel = HomeVM()
+    private var nowPlayingMovieList = [Movie]()
+    private var popularMovieList = [Movie]()
+    private var homeHeader: HomeHeader?
+    
+    var coordinator: TabsCoordinatoor?
+    
+    // MARK: - Life Cycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpViews()
+    }
+    
+    // MARK: - Private Methods
+    private func setUpViews() {
+        view.backgroundColor = .clear
+        bindViewModel()
+        setHeaderView()
+        
+        tblHome.delegate = self
+        tblHome.dataSource = self
+        tblHome.sectionHeaderTopPadding = 0
+        
+        tblHome.register(UINib(nibName: R.nib.homeTableCell.identifier, bundle: nil), forCellReuseIdentifier: R.nib.homeTableCell.identifier)
+        tblHome.register(UINib(nibName: R.nib.sectionHeader.name, bundle: nil), forHeaderFooterViewReuseIdentifier: R.nib.sectionHeader.name)
+    }
+    
+    // MARK: - Set Header
+    private func setHeaderView() {
+        homeHeader = Bundle.main.loadNibNamed(R.nib.homeHeader.name, owner: nil)?.first as? HomeHeader
+        tblHome.tableHeaderView = homeHeader
+    }
+    
+    // MARK: - Bind ViewModel
+    private func bindViewModel() {
+        viewModel.getNowPlayingMovies()
+        viewModel.getPopularMovies()
+
+        viewModel.nowPlayingMovieSuccess.bind({ [weak self] result in
+            guard let self = self else { return }
+            guard let result else { return }
+            self.nowPlayingMovieList = result
+            self.homeHeader?.submitList(list: self.nowPlayingMovieList)
+            self.tblHome.reloadData()
+        })
+        
+        viewModel.nowPlayingMovieError.bind({ error in
+            print(error)
+        })
+        
+        viewModel.popularMovieSuccess.bind({ [weak self] result in
+            guard let self = self else { return }
+            guard let result else { return }
+            self.popularMovieList = result
+            self.tblHome.reloadData()
+        })
+        
+        viewModel.popularMovieError.bind({ error in
+            print(error)
+        })
+    }
+    
+    // MARK: - Set Table Header Height
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        sizeHeaderToFit()
+    }
+
+    func sizeHeaderToFit() {
+        guard let headerView = tblHome.tableHeaderView else { return }
+
+        headerView.setNeedsLayout()
+        headerView.layoutIfNeeded()
+
+        let height = headerView.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
+        var frame = headerView.frame
+        frame.size.height = height
+        headerView.frame = frame
+
+        tblHome.tableHeaderView = headerView
+    }
+    
+}
+
+// MARK: - Extension HomeVC
```

Comment:
add extension is for what in mark // MARK: - Tableview Delegate and Datasource Extension